sympy/physics/mechanics/kane.py

```diff
@@ -227,19 +246,25 @@ def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
         self._forcelist = forcelist
         self._bodylist = bodies
 
+        self._lin_vel_constr = velocity_constraints
+        self._nonlin_vel_constr = complex_constraints
```

Comment:
The word "complex" in SymPy makes me think of complex value numbers (imaginary). You use "nonlin" here also, can we be consistent and just use nonlinear_velocity_constraints or something similar?

Reply:
> The word "complex" in SymPy makes me think of complex value numbers (imaginary). You use "nonlin" here also, can we be consistent and just use nonlinear_velocity_constraints or something similar?

I used it because Roithmayr uses it. I can change this easily.

Reply:
Given a set of m constraint equations of which some are linear in the u's and some are nonlinear in the u's, could we parse that list and seperate into linear and nonlinear equations and identify which u's are not linear coefficients, then form the respective matrices for these two groups? (or something similar). It would just be ideal if a user could pass all motion constraints to `velocity_constraints` regardless of linearity and we do all the right things internally.

Reply:
> Given a set of m constraint equations of which some are linear in the u's and some are nonlinear in the u's, could we parse that list and seperate into linear and nonlinear equations and identify which u's are not linear coefficients, then form the respective matrices for these two groups? (or something similar). It would just be ideal if a user could pass all motion constraints to `velocity_constraints` regardless of linearity and we do all the right things internally.

This should be possible! I do not think, we can treat linear and non-linear constraints the same way, but this parsing should be possible. But the issue of the ordering of the generalized coordinates would remain. If the uses gives a wild mx of linear / nonlinear motion constrains, we cannot tell him how to order the gen. coordinates.